languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/client/files_producers/ClientFileProducer.kt

```diff
@@ -160,7 +160,8 @@ export function buildRelativeUri(commonRequest: ClientRequest): string {
   var uri: string = commonRequest.uriTemplate as string
 
   for (const param in pathMap) {
-    uri = uri.replace(`{${'$'}{param}}`, `${'$'}{pathMap[param]}`)
+    const value = encodeURIComponent(`${'$'}{pathMap[param]}`)
+    uri = uri.replace(`{${'$'}{param}}`, `${'$'}{value}`)
```

Comment:
im little bit doubtful here, cause thats not backward compatibility example user is already passing encoded uri, meaning `1234%5BP%5D` will be `1234%255BP%255D`.
maybe have major version change or at least pass it as optional cmd option while init the generator sth like `--enable_encoding_uri`

@jenschude wdyt?